### PR TITLE
Upgrade to .net 6.0

### PR DESCRIPTION
**Pull request by solidcloudio**
I upgraded this to .Net 6.0 and the .Net SDK 6.0.309, One code change was required relating to X509Certificate private key. 

I also enabled builds in Configuration Manager for most of the projects.  Why were they disabled?

Compiles, All unit tests passing.

### CODE REVIEW

**Comment by solidcloudio** [comment]
@microsoft-github-policy-service agree [company="Solid Cloud, LLC"]



________________________________

**Comment by solidcloudio** [comment]
@microsoft-github-policy-service agree company="Solid Cloud, LLC"

________________________________

**Comment by egaribay77** [comment]
@solidcloudio We have been tracking this PR and were planning to ack this very soon. Is there a reason you closed this? I am reopening for now

**Comment by solidcloudio** [comment]
I read a comment somewhere that it wasn't necessary.  https://github.com/microsoft/service-fabric-yarp/issues/7
And it felt like a Ghost town around here :/

File: src/YarpProxy.Service/Service/Security/ServerCertificateBinding/SniServerCertificateSelector.cs [hunk 1]
```diff
@@ -216,7 +216,7 @@ private bool HasAccessToPrivateKey(X509Certificate2 certificate)
                 // However based on https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.pkcs.signedcms.computesignature?view=netframework-4.6.2,
                 // it seems that this can result in pin prompt even we set silent to true for some situation.
                 // This could be a potential issue which can potentially result in outage. We choose to access PrivateKey directly for verification.
-                return certificate.PrivateKey != null;
+                return certificate.HasPrivateKey;
```
Comment (davidni) [inline src/YarpProxy.Service/Service/Security/ServerCertificateBinding/SniServerCertificateSelector.cs:219]: Sugg something like:

```cs
using var key = certificate.GetRSAPrivateKey();
return key != null;
```

This isn't strictly the same as what you have here, and help verify that the current user actually can access the private key, not just that the key exists.

Comment (solidcloudio) [reply src/YarpProxy.Service/Service/Security/ServerCertificateBinding/SniServerCertificateSelector.cs:219]: Well, you know more about the sdk than I do, go ahead and fix it ;)

**Comment by egaribay77** [comment]
@solidcloudio Can you update the README https://github.com/microsoft/service-fabric-yarp/blob/main/README.md#yarp-reverse-proxy-for-service-fabric-integration with the latest versions?

**Comment by solidcloudio** [comment]
See - https://github.com/microsoft/service-fabric-yarp/pull/19

**Pull request closed without merging**